logstash_async/memory_cache.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# This software may be modified and distributed under the terms
+# of the MIT license.  See the LICENSE file for details.
+
+import uuid
+from datetime import datetime, timedelta
+from logstash_async.cache import Cache
+
+
+class MemoryCache(Cache):
+    """Backend implementation for python-logstash-async. Keeps messages in a local, in-memory cache
+    while attempting to publish them to logstash. Does not persist through process restarts. Also,
+    does not write to disk.
+
+    :param cache: Usually just an empty dictionary
+    :param event_ttl: Optional parameter used to expire events in the cache after a time
+    """
+
+    def __init__(self, cache, event_ttl=None):
+        self._cache = cache
+        self._event_ttl = event_ttl
+
+    def add_event(self, event):
+        event_id = uuid.uuid4()
+        self._cache[event_id] = {
+            "event_text": event,
+            "pending_delete": False,
+            "entry_date": datetime.now(),
+            "id": event_id
+        }
+
+    def get_queued_events(self):
+        events = []
+        for event in self._cache.values():
+            if not event['pending_delete']:
+                events.append(event)
```

Comment:
Correct me if I'm wrong: I think here is the pending_delete marker missing, like `event['pending_delete'] = True`, otherwise events will never be marked as pending_delete.

While this marker is less relevant for the in-memory cache, I think it would make sense to maintain it here anyway since also the rest of the in-memory cache maintains it as well and it should not be that relevant on performance.